app/models/step_by_step_page.rb

```diff
@@ -41,6 +41,12 @@ def self.validate_redirect(redirect_url)
     redirect_url =~ regex
   end
 
+  def status
+    return unpublished_status if has_draft? && has_been_published?
```

Comment:
This looks good 💯 

I would fix-up the bits of this commit that get and show the status to the previous commit. That way this commit only has changes related to adding the filter.

It will make the code easier to trace later on when we'll invariably have to update it.